AirportTicketBookingSystem.Tests/DomainTests/RepositoriesTests/BookingRepositoryTests.cs

```diff
@@ -0,0 +1,212 @@
+        Action act = () => _bookingRepository.CancelBooking(999);
+        act.Should().Throw<KeyNotFoundException>().WithMessage(string.Format(ErrorMessages.BookingNotFound, 999));
+    }
+
+    [Fact]
+    public void GetAllBookings_WhenCalled_ShouldReturnAllBookings()
+    {
+        var booking1 = _fixture.Create<Booking>();
+        var booking2 = _fixture.Create<Booking>();
+        _bookingRepository.AddBooking(booking1);
+        _bookingRepository.AddBooking(booking2);
+
+
+        var result = _bookingRepository.GetAllBookings();
+
+        result.Should().HaveCount(2).And.Contain(booking1).And.Contain(booking2);
+    }
+
+    [Fact]
+    public void GetAllBookings_WhenNoBookingsExist_ShouldReturnEmptyList()
+    {
+        var result = _bookingRepository.GetAllBookings();
+        result.Should().BeEmpty();
+    }
+
+    [Fact]
+    public void GetLastBooking_WhenBookingsExist_ShouldReturnLastBooking()
+    {
+        var booking1 = _fixture.Create<Booking>();
+        var booking2 = _fixture.Create<Booking>();
+        var booking3 = _fixture.Create<Booking>();
+        _bookingRepository.AddBooking(booking1);
+        _bookingRepository.AddBooking(booking2);
+        _bookingRepository.AddBooking(booking3);
+
+
+        var result = _bookingRepository.GetLastBooking();
+        result.Should().Be(booking3);
+    }
+
+    [Fact]
+    public void GetLastBooking_WhenNoBookingsExist_ShouldReturnNull()
+    {
+        var result = _bookingRepository.GetLastBooking();
+        result.ShouldBeNull();
+    }
+
+    [Fact]
+    public void UpdateBooking_WhenBookingExists_ShouldModifyExistingBooking()
+    {
+        var booking = _fixture.Build<Booking>()
+                      .With(b => b.Id, 1).Create();
+
+        var updatedFlight = _fixture.Build<Flight>()
+       .With(f => f.Id, 7).Create();
+
+        var updatedBooking = _fixture.Build<Booking>()
+                      .With(b => b.Id, 1)
+                      .With(b => b.SeatClass, SeatClass.FirstClass)
+                      .With(b => b.Flight, updatedFlight).Create();
+
+        _bookingRepository.AddBooking(booking);
+        _bookingRepository.UpdateBooking(updatedBooking);
+
+        var result = _bookingRepository.GetBookingById(booking.Id);
+
+        result.Flight.Id.Should().Be(updatedBooking.Flight.Id);
+        result.SeatClass.Should().Be(updatedBooking.SeatClass);
+    }
+
+    [Fact]
+    public void UpdateBooking_WhenBookingNotFound_ShouldThrowException()
+    {
+        var updatedBooking = _fixture.Build<Booking>()
+                     .With(b => b.Id, 999).Create();
+
+        Action act = () => _bookingRepository.UpdateBooking(updatedBooking);
+
+        act.Should().Throw<KeyNotFoundException>()
+            .WithMessage(string.Format(ErrorMessages.BookingNotFound, updatedBooking.Id));
+    }
+
+    public void Dispose()
+    {
+        if (_fileStorage is FakeFileHandler fakeFileHandler)
+        {
+            fakeFileHandler.DeleteFile("test-file");
```

Comment:
we don't need to have the file name passed, it's already configured in the handler